### DISCUSSION TITLE
Support coverage collection while running `next dev`

### DISCUSSION THREAD

**Discussion by jwickens** [vercel/next.js#70713] [Ideas]
### Goals

* Support E2E with comprehensive coverage report

### Background

While trying to collect coverage there are well documented ways to do this in Playwright or Cypress for code executed on the client. Using a tool like `nyc` we can run `nyc next dev` and this will collect coverage from the server. However for SSR code there appears to be no coverage. 

Let's say we have a page like `src/pages/home/index.html`. If we don't visit home in a test we get no coverage for that page with nyc. If we do visit it, all the top level statements are shown as covered, but not inside the functions like the page component or `getStaticProps`. 

### Proposal

I imagine the next server renders the page in a separate process. It would be great if there was an integration with nyc so that coverage is still collected for server side page renders.

**Comment by kylemh** [comment]
In general, some sort of solution to https://github.com/vercel/next.js/discussions/70713 would be huge for larger companies deciding to use API routes or request-time, server-side Next.js in App router apps. We have no ability to generate coverage reports for that code!
